solsys_code/views.py

```diff
@@ -1,3 +1,830 @@
+    # XXX TODO can almost certainly get this back out of `pointing` - technically yes
+    # but it requires including most of `apco` and `apcs`, minus the part that does the
+    # geocenter->observer corrections, in here.
+    jd1_tdb, jd2_tdb = get_jd12(obstime, 'tdb')
+    earth_pv_heliocentric, earth_pv = erfa.epv00(jd1_tdb, jd2_tdb)
+    earth_heliocentric = earth_pv_heliocentric['p']
+    # refraction constants
+    refa, refb = 0.0, 0.0  # airless apparent (for now) #_refco(frame_or_coord)
+
+    return erfa.apco(
+        jd1_tdb,
+        jd2_tdb,
+        earth_pv,
+        earth_heliocentric,
+        x,
+        y,
+        s,
+        era,
+        lon,
+        lat,
+        height,
+        xp,
+        yp,
+        sp,
+        refa,
+        refb,
+    )
+
+
+def split_number_unit_regex(s):
+    """
+    Matches a number (integer or float) followed by an optional unit
+    """
+
+    match = re.match(r'([-+]?\d*\.?\d+)([a-zA-Z%]+)?', s)
+    if match:
+        number = float(match.group(1))  # Convert to float for numerical operations
+        unit = match.group(2) if match.group(2) else ''  # Handle cases with no unit
+        return number, unit
+    else:
+        return None, None
+
+
+class MakeEphemerisView(FormView):
+    """
+    View for making an ephemeris
+    """
+
+    template_name = 'ephem_form.html'
+    form_class = EphemerisForm
+
+    def get_target_id(self):
+        """
+        Parses the target id for the given observation from the query parameters.
+
+        Returns
+        -------
+        int
+            id (primary key) of the target for ephemeris generation
+        """
+
+        if self.request.method == 'GET':
+            return self.kwargs['pk']
+        elif self.request.method == 'POST':
+            return self.request.POST.get('target_id')
+
+    def get_initial(self):
+        """
+        Populate form's HiddenField with the target_id
+        """
+        initial = super().get_initial()
+        if not self.get_target_id():
+            raise Exception('Must provide target_id')
+        target_id = self.get_target_id()
+
+        initial['target_id'] = target_id
+        initial.update(self.request.GET.dict())
+        return initial
+
+    def get_context_data(self, **kwargs):
+        """
+        Extract the pk from the kwargs and get the Target and add it to the context.
+        """
+        context = super().get_context_data(**kwargs)
+
+        target_id = self.kwargs['pk']
+        context['target'] = Target.objects.get(id=target_id)
+
+        return context
+
+    def get_form(self, form_class=None):
+        """
+        Form handler
+        """
+        form = super().get_form()
+        target_id = self.get_target_id()
+
+        cancel_url = reverse('home')
+        if target_id:
+            cancel_url = reverse('tom_targets:detail', kwargs={'pk': target_id})  # + '?tab=ephemeris'
+        form.helper.layout = Layout(
+            HTML(
```

Comment:
This workflow doesn't make much sense. 
As above, I would change the MPC code to a choice field and link to the observatory creation page in the help text. You could have a "New" option that revealed a place to enter a new facility if you like, but this workflow is still broken. 

As is, I'm taken out of the ephem creation process to make my observatory and have to start over. This would be better if you just took whatever they gave you, redirected to a confirmation page asking if they wanted to add a new observatory, then save it on the backend and redirect back to the ephem list using the new facility.